lib/PersistHelper.ts

```diff
@@ -1001,3 +1019,181 @@ export const retrieveLoginMessageSentStatus = async ({
 
     return result[0].isLoginMessageSent;
 };
+
+type LastBridgedMessageInfo = {
+    senderId: string,
+    text: string,
+    roomId: string,
+    fileName: string,
+}
+
+export const getLastBridgedMessage = async ({
+    read,
+    rocketChatUserId,
+}: {
+    read: IRead;
+    rocketChatUserId: string;
+}): Promise<LastBridgedMessageInfo | null> => {
+    const associations: Array<RocketChatAssociationRecord> = [
+        new RocketChatAssociationRecord(
+            RocketChatAssociationModel.MISC,
+            MiscKeys.BridgedMessage
+        ),
+        new RocketChatAssociationRecord(
+            RocketChatAssociationModel.USER,
+            rocketChatUserId
+        ),
+    ];
+
+    const persistenceRead: IPersistenceRead = read.getPersistenceReader();
+
+    const result = (await persistenceRead.readByAssociations(
+        associations
+    )) as unknown as Array<LastBridgedMessageInfo>;
+
+    if (!result) {
+        return null;
+    }
+
+    return result[0];
+};
+
+const simpleHash = (input: string): string => sha256(input).toString()
+
+const calculatePriorityDate = (message: IMessage) => {
+    const currentDate = new Date();
+    let priorityDate = currentDate;
+
+    if (message.updatedAt) {
+      const updatedAtDate = new Date(message.updatedAt);
+      if (!isNaN(updatedAtDate.getTime())) {
+        priorityDate = updatedAtDate;
+      }
+    } else if (message.createdAt) {
+      const createdAtDate = new Date(message.createdAt);
+      if (!isNaN(createdAtDate.getTime())) {
+        priorityDate = createdAtDate;
+      }
+    }
+
+    return priorityDate;
+  }
+
+export const generateMessageFootprint = (message: IMessage, room: IRoom, sender: IUser): string => {
+    const dateObject = calculatePriorityDate(message);
+    dateObject.setSeconds(0);
+    dateObject.setMilliseconds(0);
+
+    const timestamp = dateObject.getTime();
+    const messageProperties = `${message.text}${timestamp}`;
+    const roomProperties = `${room.id}${room.type}${room.displayName}`;
+
+    const senderProperties = `${sender.id}${sender.username}`;
+
+    const combinedProperties = `${messageProperties}${roomProperties}${senderProperties}`;
+
+    return simpleHash(combinedProperties);
+}
+
+export const saveLastBridgedMessageFootprint = async ({
+    persistence,
+    rocketChatUserId,
+    messageFootprint,
+    // source
+}: {
```

Comment:
please avoid adding types later as it increases the number of lines and complexity of the file.